collateral.md

```diff
@@ -26,6 +32,63 @@ module COLLATERAL
         </flips>
       </collateral>
 
+    syntax MCDContract ::= GemContract
+    syntax GemContract ::= "Gem" Address
+    syntax MCDStep ::= GemContract "." GemStep [klabel(gemStep)]
+ // ------------------------------------------------------------
+    rule contract(Gem ACCTGEM . _) => Gem ACCTGEM
+    rule [[ address(Gem ACCTGEM) => ACCTGEM ]] <gem-addr> ACCTGEM </gem-addr>
+
+    syntax GemAuthStep
+    syntax GemStep ::= GemAuthStep
+    syntax AuthStep ::= GemContract "." GemAuthStep [klabel(gemStep)]
+ // -----------------------------------------------------------------
+    rule <k> Gem _ . _ => exception ... </k> [owise]
+
+    syntax GemStep ::= "transferFrom" Address Address Int
+ // -----------------------------------------------------
+    rule <k> Gem ACCTGEM . transferFrom ACCTSRC ACCTDST VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Int VALUE )
+             ACCTDST |-> ( BALANCE_DST => BALANCE_DST +Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires BALANCE_SRC >=Int VALUE
+
+    syntax GemStep ::= "move" Address Address Int
+ // ---------------------------------------------
+    rule Gem ACCTGEM . (move ACCTSRC ACCTDST VALUE => transferFrom ACCTSRC ACCTDST VALUE)
```

Comment:
```suggestion
    rule <k> Gem ACCTGEM . (move ACCTSRC ACCTDST VALUE => transferFrom ACCTSRC ACCTDST VALUE) ... </k>
```

Either wrap in `<k>` tags or make a `macro` rule that looks like this:

```
rule move ACCTSRC ACCTDST VALUE => transferFrom ACCTSRC ACCTDST VALUE [macro]
```

I don't like the implicit completion of `<k> ... </k>` that happens when you don't wrap in `<k>` tags explicitly.

Reply:
But I would choose to wrap in `<k> ... </k>` tags since the other rules do too.